dolphinscheduler-task-plugin/dolphinscheduler-task-aliyunserverlessspark/src/main/java/org/apache/dolphinscheduler/plugin/task/aliyunserverlessspark/AliyunServerlessSparkTask.java

```diff
@@ -0,0 +1,236 @@
+            Map<String, String> headers = new HashMap<>();
+            log.info("[debug111] aliyunServerlessSparkParameters - {}", aliyunServerlessSparkParameters);
+            StartJobRunResponse startJobRunResponse = aliyunServerlessSparkClient.startJobRunWithOptions(
+                    aliyunServerlessSparkParameters.getWorkspaceId(), startJobRunRequest, headers, runtime);
+            jobRunId = startJobRunResponse.getBody().getJobRunId();
+            setAppIds(jobRunId);
+            log.info("Successfully submitted serverless spark job, jobRunId - {}", jobRunId);
+
+            while (!RunState.isFinal(currentState)) {
+                GetJobRunRequest getJobRunRequest = buildGetJobRunRequest();
+                GetJobRunResponse getJobRunResponse = aliyunServerlessSparkClient
+                        .getJobRun(aliyunServerlessSparkParameters.getWorkspaceId(), jobRunId, getJobRunRequest);
+                currentState = RunState.valueOf(getJobRunResponse.getBody().getJobRun().getState());
+                log.info("job - {} state - {}", jobRunId, currentState);
+                Thread.sleep(10 * 1000L);
+            }
+
+            setExitStatusCode(mapFinalStateToExitCode(currentState));
+
+        } catch (Exception e) {
+            log.error("Serverless spark job failed!", e);
+            throw new AliyunServerlessSparkTaskException("Serverless spark job failed!");
+        }
+    }
+
+    @Override
+    public void submitApplication() throws TaskException {
+
+    }
+
+    @Override
+    public void trackApplicationStatus() throws TaskException {
+
+    }
+
+    protected int mapFinalStateToExitCode(RunState state) {
+        switch (state) {
+            case Success:
+                return TaskConstants.EXIT_CODE_SUCCESS;
+            case Failed:
+                return TaskConstants.EXIT_CODE_KILL;
+            default:
+                return TaskConstants.EXIT_CODE_FAILURE;
+        }
+    }
+
+    @Override
+    public AbstractParameters getParameters() {
+        return aliyunServerlessSparkParameters;
+    }
+
+    @Override
+    public void cancelApplication() throws TaskException {
+        CancelJobRunRequest cancelJobRunRequest = buildCancelJobRunRequest();
+        try {
+            aliyunServerlessSparkClient.cancelJobRun(aliyunServerlessSparkParameters.getWorkspaceId(), jobRunId,
+                    cancelJobRunRequest);
+        } catch (Exception e) {
+            log.error("Failed to cancel serverless spark job run", e);
+        }
+    }
+
+    @Override
+    public List<String> getApplicationIds() throws TaskException {
+        return Collections.emptyList();
+    }
+
+    protected Client buildAliyunServerlessSparkClient(String accessKeyId, String accessKeySecret,
+                                                      String regionId) throws Exception {
+        String endpoint = String.format("emr-serverless-spark.%s.aliyuncs.com", regionId);
```

Comment:
I have seen this string in some other places, maybe a constant is more suitable.

Reply:
Sure, will fix it.

Reply:
@lenboo Fixed : )